plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java

```diff
@@ -235,12 +238,18 @@ public ConnectorSplitSource getSplits(
             return emptySplitSource();
         }
 
+        Set<String> neededColumnNames = Streams.concat(hiveTable.getProjectedColumns().stream(), hiveTable.getConstraintColumns().stream())
```

Comment:
And also predicated on?

Reply:
WDYM?
```
hiveTable.getEnforcedConstraint().getDomains().map(Map::keySet).orElse(ImmutableSet.of()).stream()
```

Reply:
actually this one is not needed and neither unenforced is. 
nvrm